### PR TITLE
Create common base class and update docs for AnP config-adding commands

### PR DESCRIPTION
**Pull request by tmooney**
The new base class should help keep the docs and functionality in synch going forward!

### CODE REVIEW

File: lib/perl/Genome/Config/AnalysisProject/Command/AddConfigBase.pm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+sub valid_statuses {
+    return ("Pending", "Hold", "In Progress", "Template");
+}
+
+sub execute {
+    my $self = shift;
+
+    my @new_items = $self->_create_profile_items();
```
Comment (nnutter) [inline lib/perl/Genome/Config/AnalysisProject/Command/AddConfigBase.pm]: Should we create a `_create_profile_items()` stub?

File: lib/perl/Genome/Config/AnalysisProject/Command/AddModel.t [hunk 2]
```diff
@@ -104,7 +104,9 @@ EOFILE
         store_only => 1,
         reprocess_existing => 0,
     );
-    my $profile_item = $add_cmd->execute();
+    my $ok = $add_cmd->execute();
+    ok($ok, 'executed add command');
+    my $profile_item = Genome::Config::Profile::Item->get(analysis_project => $analysis_project, status => 'disabled');
```
Comment (nnutter) [inline lib/perl/Genome/Config/AnalysisProject/Command/AddModel.t]: Should we verify `$profile_item` doesn't exist before the `$add_cmd->execute()`?

**Comment by nnutter** [comment]
:+1:

**Comment by acoffman** [comment]
+1

It does seem a little odd that the `is_many` input is called `tag` rather than `tags`. It makes the line 
````perl
for my $tag ($self->tag) {}
```` 
read a little oddly. It looks like the singular naming is left over from the previous versions of the commands, so perhaps its best not to break compatibility. I'm fine with it either way!

File: lib/perl/Genome/Config/AnalysisProject/Command/AddConfigBase.pm (hunk 1 above)
Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/Command/AddConfigBase.pm]: Sure! I've added one.

File: lib/perl/Genome/Config/AnalysisProject/Command/AddModel.t (hunk 2 above)
Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/Command/AddModel.t]: Sounds like a good idea--added!

**Comment by tmooney** [comment]
@acoffman You're not the only one who thinks `--tag` should be `--tags`!  (See also AUE-427.) That's what motivated this PR originally :smile:

I was thinking that if consensus was to change it (which it sounds like it is) this would be followed up with a second PR that made that change in its new central location.

**Pull request merged**